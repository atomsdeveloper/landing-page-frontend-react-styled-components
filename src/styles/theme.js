export const theme = {
    colors: {
        primaryColor: 'blue',
        secondaryColor: 'red'
    },
    fonts: {
        
    },
    spacings: {
        
    }
}