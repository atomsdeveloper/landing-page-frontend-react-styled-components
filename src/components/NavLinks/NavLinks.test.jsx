import { screen } from '@testing-library/react';
import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import theme from '../../styles/theme';

import { NavLinks } from '.';

import mock from './mock';

describe('<NavLinks />', () => {
  it('should render links', () => {
    render(<ThemeProvider theme={theme}> <NavLinks links={mock} /> </ThemeProvider>);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    render(<ThemeProvider theme={theme}> <NavLinks /> </ThemeProvider>);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });

  it('should render links', () => {
    render(<ThemeProvider theme={theme}> <NavLinks links={mock} /> </ThemeProvider>);
    expect(screen.getByText(/link 10/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
    );
  });

  it('should match snapshot', () => {
    render(<ThemeProvider theme={theme}> <NavLinks links={mock} /> </ThemeProvider>);
    expect(screen.getAllByRole('link')).toMatchSnapshot();
  });
});