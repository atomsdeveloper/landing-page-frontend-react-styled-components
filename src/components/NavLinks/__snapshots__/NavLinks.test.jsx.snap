// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c0 {
  display: flex;
  flex-flow: row wrap;
}

@media (max-width: 768px) {
  .c0 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<nav
  class="c0"
/>
`;
